### PR TITLE
add quantize, save, load function for transformers-like api

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change
optimum-intel PR:  https://github.com/huggingface/optimum-intel/pull/880
example code:
```
from transformers import AutoTokenizer
from optimum.intel import INCModelForCausalLM
from neural_compressor.transformers import GPTQConfig, RtnConfig
quantization_config = GPTQConfig(
    bits=4, sym=True, damp_percent=0.01, desc_act=True
)
user_model = INCModelForCausalLM.from_pretrained(self.model_name, quantization_config=quantization_config)
output = user_model(self.input_ids)
user_model.save_pretrained("tmp_gptq")
loaded_model = INCModelForCausalLM.from_pretrained("tmp_gptq")
loaded_output = loaded_model(self.input_ids)
assert torch.allclose(output, loaded_output, atol=1e-2), "Compare failed!"
```

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by changwangss** [comment]
code coverage drop > 1%, #1987 will improve it. @XuehaoSun @chensuyue

File: neural_compressor/transformers/models/modeling_auto.py [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+                    zp=zp,
+                    bias=m.bias is not None,
+                    g_idx=True,
+                    use_optimum_format=use_optimum_format,
+                )
+            set_module(model, n, new_module)
+    return model
+
+
+class _BaseQBitsAutoModelClass:
```
Comment (thuang6) [inline neural_compressor/transformers/models/modeling_auto.py:79]: do we still need support for QBit? or here is just reused the original name from ITREX but no real connection with QBits?  if so, suggest to change to a proper name

**Review by thuang6 — APPROVED** [review]

File: neural_compressor/transformers/models/modeling_auto.py (hunk 1 above)
Comment (changwangss) [reply neural_compressor/transformers/models/modeling_auto.py:79]: yes, it is just reuse the original name, improve it to _BaseINCAutoModelClass.

**Comment by thuang6** [comment]
> CI failed due to code coverage drop > 1%, #1987 will improve it. @XuehaoSun @chensuyue

Chang conducted local test based on her PR branch on Optimum-Intel but without adding that test to UT which will introduce Optimum-Intel dependency,  how about we get this PR merged for patch release and improve in later PR for transformer-like API @chensuyue

**Review by chensuyue — APPROVED** [review]

**Pull request merged**